test-data/unit/check-dataclasses.test

```diff
@@ -2036,4 +2035,38 @@ Foo(
     present_4=4,
     present_5=5,
 )
+
+[builtins fixtures/dataclasses.pyi]
+
+[case testReplace]
+from dataclasses import dataclass, replace, InitVar
+from typing import ClassVar
+
+@dataclass
+class A:
+    x: int
+    q: InitVar[int]
+    q2: InitVar[int] = 0
+    c: ClassVar[int]
+
+
+a = A(x=42, q=7)
+a2 = replace(a)  # E: Missing named argument "q" for "replace" of "A"
+a2 = replace(a, q=42)
+a2 = replace(a, x=42, q=42)
+a2 = replace(a, x=42, q=42, c=7)  # E: Unexpected keyword argument "c" for "replace" of "A"
+a2 = replace(a, x='42', q=42)  # E: Argument "x" to "replace" of "A" has incompatible type "str"; expected "int"
+a2 = replace(a, q='42')  # E: Argument "q" to "replace" of "A" has incompatible type "str"; expected "int"
+reveal_type(a2)  # N: Revealed type is "__main__.A"
+
 [builtins fixtures/dataclasses.pyi]
+
+[case testReplaceNotDataclass]
+from dataclasses import replace
+
+replace(5)  # E: Argument 1 to "replace" has incompatible type "int"; expected a dataclass
+
+class C:
+    pass
+
+replace(C())  # E: Argument 1 to "replace" has incompatible type "C"; expected a dataclass
```

Comment:
I would propose to add tests that cover:
* Generic dataclasses
* Fine-grained mode (to check a call elsewhere will be re-checked on field type change)

Reply:
To clarify on the fine grained (daemon) mode, I want to double-check that if there is only a replace call in a different module (and nothing else), it will be correctly rechecked (e.g. a new error will appear) if you change one of the dataclass field types.

Reply:
Good idea, did I do this right? d71bc21f15a98ad356461d53494a8b9c0ab0f17e